core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -238,127 +239,136 @@ trait Endpoint[A] { self =>
 
   /**
-   * Validates the result of this endpoint using a predefined `rule`. This method allows for rules
-   * to be reused across multiple endpoints.
-   *
-   * @param rule the predefined [[ValidationRule]] that will return true if the data is
-   *             valid
-   *
-   * @return an endpoint that will return the value of this reader if it is valid.
-   *         Otherwise the future fails with an [[Error.NotValid]] error.
-   */
-  final def should(rule: ValidationRule[A]): Endpoint[A] = should(rule.description)(rule.apply)
+    * Validates the result of this endpoint using a predefined `rule`. This method allows for rules
+    * to be reused across multiple endpoints.
+    *
+    * @param rule the predefined [[ValidationRule]] that will return true if the data is
+    *             valid
+    *
+    * @return an endpoint that will return the value of this reader if it is valid.
+    *         Otherwise the future fails with an [[Error.NotValid]] error.
+    */
+  final def should(rule: ValidationRule[A]): Endpoint[F, A] = should(rule.description)(rule.apply)
 
   /**
-   * Validates the result of this endpoint using a predefined `rule`. This method allows for rules
-   * to be reused across multiple endpoints.
-   *
-   * @param rule the predefined [[ValidationRule]] that will return false if the data is
-   *             valid
-   *
-   * @return an endpoint that will return the value of this reader if it is valid.
-   *         Otherwise the future fails with a [[Error.NotValid]] error.
-   */
-  final def shouldNot(rule: ValidationRule[A]): Endpoint[A] = shouldNot(rule.description)(rule.apply)
+    * Validates the result of this endpoint using a predefined `rule`. This method allows for rules
+    * to be reused across multiple endpoints.
+    *
+    * @param rule the predefined [[ValidationRule]] that will return false if the data is
+    *             valid
+    *
+    * @return an endpoint that will return the value of this reader if it is valid.
+    *         Otherwise the future fails with a [[Error.NotValid]] error.
+    */
+  final def shouldNot(rule: ValidationRule[A]): Endpoint[F, A] = shouldNot(rule.description)(rule.apply)
 
   /**
     * Lifts this endpoint into one that always succeeds, with [[Try]] representing both success and
     * failure cases.
     */
-  final def liftToTry: Endpoint[Try[A]] =
-    new Endpoint[Try[A]] with (Try[Output[A]] => Output[Try[A]]) {
-      final def apply(toa: Try[Output[A]]): Output[Try[A]] = toa match {
-        case Return(oo) => oo.map(Return.apply)
-        case t @ Throw(_) => Output.payload(t.cast[A])
+  final def liftToTry: Endpoint[F, Try[A]] = attempt.map({
+    case Right(r) => Return(r)
+    case Left(t) => Throw(t)
+  })
+
+  /**
+    * Lifts this endpoint into one that always succeeds, with [[Either[Throwable, A]] representing both success and
+    * failure cases.
+    */
+  final def attempt: Endpoint[F, Either[Throwable, A]] =
```

Comment:
Side note: I support the idea of renaming this. I think we can even do that outside of this PR.